src/com/google/javascript/jscomp/NodeUtil.java

```diff
@@ -2362,6 +2362,14 @@ public static boolean isExprCall(Node n) {
         && n.getFirstChild().isCall();
   }
 
+  static boolean isIIFE(Node n) {
+    Node parent = n.getParent();
+    return n.isFunction()
+        && parent.isCall()
+        && parent.getFirstChild() == n
+        && parent.getParent().isExprResult();
```

Comment:
For my own information, why does it matter if the grandparent is an ExprResult?
Perhaps a comment giving a bad case we're trying to avoid would make this clearer.